pkg/monitoring/domainstats/downwardmetrics/scraper.go

```diff
@@ -30,25 +31,49 @@ type StaticHostMetrics struct {
 }
 
 type Scraper struct {
-	isolation            isolation.PodIsolationDetector
-	staticHostInfo       *StaticHostMetrics
-	hostMetricsCollector *hostMetricsCollector
+	isolation isolation.PodIsolationDetector
+	reporter  *DownwardMetricsReporter
 }
 
 func (s *Scraper) Scrape(socketFile string, vmi *k6sv1.VirtualMachineInstance) {
 	if !vmi.IsRunning() || !downwardmetrics.HasDownwardMetricDisk(vmi) {
 		return
 	}
 
+	metrics, err := s.reporter.Report(socketFile)
```

Comment:
The `Scrape()` and `Report()` are part of the old method of sharing the metrics using a disk, that should be deprecated in future release since vhostmd did it (and SAP will not support it anymore). So, I don't think it's worth adding a test for it.

I was also confused by how the scraper call collector, and collector calls scrapper (I think is to avoid cyclic imports), but again, as soon the downwardmetrics disk transport gets deprecated we can remove all that code. 

The `Report()` method was not only collecting the metrics but also written them to disk, so to re-use the code that collects the metrics, we just moved the write-to-disk code to the `Scrape()` method. Note that, `Scrap()`is part of an interface used by prometheus, so I didn't want to touch too much of that code.

Reply:
The `Scrape` calls `Report`, this is done this way only because we reuse the concurrent collector. The `Report` on the Scraper is internal(can be unexported)

Bedore the `Report` was not collecting the metrics. I might be missing what you think by that